src/image.rs

```diff
@@ -377,12 +377,35 @@ pub(crate) fn decoder_to_vec<'a>(decoder: impl ImageDecoder<'a>) -> ImageResult<
 }
 
 /// Represents the progress of an image operation.
+///
+/// Note that this is not necessarily accurate and no change to the values passed to the progress
+/// function during decoding will be considered breaking. A decoder could in theory report the
+/// progress `(0, 0)` if progress is unknown, without violating the interface contract of the type.
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub struct Progress {
     current: u64,
     total: u64,
 }
 
+impl Progress {
```

Comment:
Instead of this impl block, I'd rather just have the two fields be marked pub.

Reply:
Not really, as explain in the commit comment I'm not sure if we want to add different fields in the future? Like you said, it *might* mean bytes decoded but it could also be pixels. Until that is clear, I'd rather have the members be private so we can switch at any point.